assignments/array-methods.js

```diff
@@ -55,29 +55,82 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
-console.log(fullName);
+
+//  Study Group Solution
+let fullName = [[runners.first_name] + [runners.last_name]];
+ let result = fullName.map(function(runners) {
+     return [runners.first_name + runners.last_name];
+ })
+ console.log(fullName);
+
+
+
+
+// Q&A Solution
+
+ let fullName=[]
+     fullName = runners.item  {
+     fullName.push('${item.first_name}, $(item.last_name}')
+     return (fullName);
+ }
+
+ console.log(fullName)
+
+
+
+
+
+
+
 
 // ==== Challenge 2: Use .map() ====
-// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
+// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. 
+// Convert each first name into all caps and log the result
+
+//  Q&A way
+  let allCaps = runnners.map(item=>item.first_name.toUpperCase());
+  console.log(allCaps);
+
+// Study Group Solution
+
 let allCaps = [];
-console.log(allCaps); 
+runners.map(upper => allCaps.push(upper.first_name.toUpperCase()));
+    console.log(allCaps); 
 
-// ==== Challenge 3: Use .filter() ====
+    // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+runners.filter( large => large.shirt_size === "L");
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
+
+
+
 let ticketPriceTotal = [];
+
+ticketPriceTotal = runners.reduce(function(ticket,item) {
+    return ticket += item.donation;
+    
+}, 0 );
```

Comment:
Great job here, I would just change the item parameter to runner, since it is what is being received and more semantic